10-JavaScript/js-intro.md

```diff
@@ -0,0 +1,439 @@
+
+Open the `node` REPL and put in the following commands. What do you get?
+
+1. `'Hello World!';`
+1. `2 + 1;`
+1. `100 * 4;`
+1. `2 + 'Hello World!';`
+1. `let x = 7;`
+1. `x + 2;`
+1. `typeof x;`
+1. `2 - 'Hello World!';`
+
+<details>
+
+  <summary>
+    Compare your answers here
+  </summary>
+
+  1. `'Hello World!'`
+  1. `3`
+  1. `400`
+  1. `'2Hello World!'`
+  1. `undefined`
+  1. `9`
+  1. `'number'`
+  1. `NaN`
+
+Keep a note of what answers are surprising. We will get into what some of these answers mean, soon.
+
+</details>
+
+### Exit the Node REPL
+
+Use `.exit` or ctrl-d to get out of the Node REPL
+
+Now that we've got a REPL available to us, let's jump in with some JavaScript specifics.
+
+## JavaScript: Code Style & Commenting
+
+> One of the first things to learn in a programming language is any naming convention and how to comment code. What are the conventions in JavaScript?
+
+> _-- Some cool programmer learning a new language_
+
+Let's learn some rules about JavaScript's syntax:
+
+1. In order to make a comment, we will use `//`
+1. The naming convention for variables is camelCase
+1. In JavaScript, the semicolon (`;`) is a consistent way to state that a line of code is ending
+
+Compare this to Ruby:
+
+| Ruby | JavaScript |
+| --- | ---
+| Comments are made with a line beginning with `#` | Comments are made with a line beginning with `//`
+| variables are named using snake_case | variables are named using camelCase
+| lines of code can end with `;` and it's valid, but nobody really does this | lines of code can end with `;`, and we will learn JavaScript best practices of doing this
+
+At some point, we may have more nuanced opinions about this; for now, let's go with this :)
+
+## Data Types in JavaScript
+
+> How can I represent data in this language? What are the data types available in this language?
+
+> _-- Some cool programmer learning a new language_
+
+JavaScript has types that are similar to Ruby:
+
+- `Boolean`
+- `Number`
+- `String`
+- `Array`
+  - technically these aren't a distinct type, but we won't get into that now
+- `null`
+
+And also some that are distinct from Ruby:
+- `undefined`
+- `Object`
+- `Function`
+  - similar to arrays, technically these aren't a distinct type, but we won't get into that now, and it's worthwhile to separate it on this list
+
+The best way for us to get into JavaScript at the moment is to blitz through all of the familiar types. Feel free to follow along in the Node REPL to verify that these these are real things in JavaScript.
+
+#### An Example Error in Node REPL
+
+If something is not real, Node REPL will let us know by _throwing an error._
+
+For example, if we have the Node REPL open and type in something that the Node REPL doesn't recognize, it may have output that looks like any of these options:
```

Comment:
Line 132 feels a little odd (what does "not real" mean?). I might rephrase to

Much like `irb` or `pry`, if the Node REPL encounters code it can't interpret, it will let us know by _throwing an error_. This could be due to a _syntax error_ (the thing you typed isn't valid JavaScript), or a _runtime error_ (you gave it valid JavaScript that contains a logic bug).